drivers/scheduler/k8s/k8s.go

```diff
@@ -6114,6 +6114,26 @@ func (k *K8s) validateCsiSnap(pvcName string, namespace string, csiSnapshot v1be
 	return nil
 }
 
+// GetPodsRestartCount return map of HostIP and it restart count in given namespace
+func (k *K8s) GetPodsRestartCount(namespace string, podLabelMap map[string]string) (map[*v1.Pod]int32, error) {
+	podRestartCountMap := make(map[*v1.Pod]int32)
+	podList, err := k8sCore.GetPods(namespace, podLabelMap)
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range podList.Items {
+		actualPod, err := k8sCore.GetPodByName(pod.Name, pod.Namespace)
+		if err != nil {
+			return nil, err
+		}
+		containerStatus := actualPod.Status.ContainerStatuses
+		for _, status := range containerStatus {
+			podRestartCountMap[actualPod] += status.RestartCount
```

Comment:
do we need all containers restart count?

Reply:
Yes. It calculate restart count collectively for all containers in a pod ..